js/RecentDonations.js

```diff
@@ -0,0 +1,196 @@
+      if (error) {
+        $scope.errors.push(problem + ": " + error)
+      } else {
+        $scope.errors.push(problem)
+      }
+      dataChanged()
+    }
+    
+    // Called whenever our data is updated by callbacks.
+    var dataChanged = function() {
+      // Only update once all our callbacks have added their data.
+      if (waitingOn == 0) {
+        
+      /* // Some test data.
+        contributions = [
+        {
+          name: "Alex Karantza",
+          date: Date.parse("07 05 1988"),
+          amount: "$100.00",
+          message: "I am the very model of a modern major general, Ive information vegetable animal and mineral; I know the kings of england and I quote the fights historical, from marathon to waterloo in order catagorical",
+        },
+        {
+          name: "Mike Stark",
+          date: Date.parse("08 01 1994"),
+          amount: "$99.00",
+          message: "DOUBLE HEAD",
+        },
+        {
+          name: "Adina Rubinoff",
+          date: Date.parse("10 18 1989"),
+          amount: "$101.00",
+          message: null,
+        },
+        ]
+        */
+
+        // Sort all our contributions
+        contributions.sort(function(a,b) { 
+          return b.date - a.date 
+        })
+    
+        var subset = contributions.slice(0, count)
+
+        // And send the latest 'count' to the UI.
+        //var subset = contributions.
+        $scope.contributions = subset
+        $scope.$apply()
+
+
+        // Now let's get a message from those contributors, if we can
+        if (subset.length > 0) {
+
+          // Starting after our current message, look and see if there's another message to display (wrapping back to the first if necessary)
+          for (var i = 1; i <= subset.length; i++) {
+            var idx = (messageBeingRead + i) % subset.length
+            var donation = subset[idx]
+            // If there is one, display it.
+            if (donation.message) { 
+              // Twitter rules.
+              var message = donation.message
+              if (message.length > 143) { 
+                message = message.substr(0,140) + "..." 
+              }
+
+              if (messageBeingRead !== idx) {
+                displayMessage (donation.name + ": " + message)
+                messageBeingRead = idx
+              }
+
+              break
+            }
+          }
+         
+        }
+      }
+    }
+
+    // Animate a new message onto the page.
+    var displayMessage = function(message) {
+      $('#message').fadeOut(500, function() { 
+        $scope.message = message
+        $scope.$apply()
+        $('#message').fadeIn(500)
+      })
+    }
+
+    // Gets the name (and IDs) of the requested user(s)
+    var getParticipantIDs = function(then) {
+      if (isTeam) { 
+        // We need the name of the team, and the list of members.
+        GLOBALS.GET_TEAM_INFO(uid, function(data) {
+          $scope.displayName = data.name || "(Invalid ID)"
```

Comment:
I suppose I should be checking if data.name == undefined, though I doubt having a blank name is allowed anyway. Unless your name is 'false' or something.